src/ray/raylet/worker_pool.cc

```diff
@@ -370,7 +371,11 @@ Process WorkerPool::StartWorkerProcess(
                 << " worker(s) with pid " << proc.GetId();
   MonitorStartingWorkerProcess(proc, language, worker_type);
   state.starting_worker_processes.emplace(
-      proc, StartingWorkerProcessInfo{workers_to_start, workers_to_start, worker_type});
+      startup_token_,
+      std::make_pair(
```

Comment:
Can you put startup_token into the StartingWorkerProcessInfo? Then we don't need an additional level of indirections :)

Reply:
The core of the fix is using `startup_token` as key. I can additionally add it in `StartingWorkerProcessInfo` and make necessary changes else where.

Reply:
Ah sorry, I meant put `proc` into `StartingWorkerProcessInfo` and get rid of the std::make_pair (startup_token should be the key yes). It mainly seems odd to use both `std::make_pair` here and also have the struct. This will simplify the code elsewhere because then you don't need the additional `first` and `second` calls.